### ISSUE TITLE
Convert cloud-provider installs to leverage docker-machine

### ISSUE THREAD

**Issue by moxiegirl** [docker/docker#18282]
For installing docker on cloud platforms we want to direct people to install machine and then deploy docker. 

@londoncalling I'd like you to do this update.  

cc @shykes

**Comment by londoncalling** [comment]
Got it. It looks like there is an install topic per Cloud provider:
* Amazon EC2: http://docs.docker.com/engine/installation/amazon/
* Google Cloud: http://docs.docker.com/engine/installation/google/
* IBM Soft Layer: http://docs.docker.com/engine/installation/softlayer/
* Microsoft Azure: http://docs.docker.com/engine/installation/azure/
* Rackspace: http://docs.docker.com/engine/installation/rackspace/
* Joyent: http://docs.docker.com/engine/installation/joyent/

Most of these are a page or less long, some with only links, and some with step-by-step instructions. Some link to explanations of using Docker Machine to install (e.g., Azure: https://azure.microsoft.com/en-us/documentation/articles/virtual-machines-docker-machine/) 

For each topic, should we remove other install instructions and re-write with specific examples per  https://docs.docker.com/machine/get-started-cloud/ ? Or Simply add a link to that "cloud" topic? I assume the former (re-write each).

**Comment by londoncalling** [comment]
@moxiegirl per our conversation:
* Figure out which cloud providers have the Docker Machine plugin that allows it to provision hosts on that provider, and list them in the docs
* Re-write cloud installs to promote use of Docker Machine to install on cloud providers (for all , with a specific example)
* Document manual install of Docker on cloud providers that don't provide the Docker Machine plugin (generic)

**Comment by thaJeztah** [comment]
Thanks @londoncalling! Looking forward to your pull request :smiley:

**Comment by londoncalling** [comment]
@nathanleclaire, fyi, per part of our conversation today, I'll be re-working the install docs w/regard to this is the issue

**Comment by nathanleclaire** [comment]
:+1:

**Comment by londoncalling** [comment]
@nathanleclaire, @moxiegirl, to review our meeting, and a status update on this:

* Before I rewrite/re-org these topics, I want to successfully install Docker without Docker Machine, then with Docker Machine on the 3 primary cloud platforms (Digital Ocean, AWS EC2, Microsoft Azure)

* I installed on all 3 platforms natively per their instructions.(SUCCESS)
* I installed a Digital Ocean "droplet"  with Docker Machine, and even uninstalled it (per https://docs.docker.com/machine/get-started-cloud/ and some of my own ingenuity) (SUCCESS)

However, for the other two cloud installs using Docker Machine, I've reached an impasse as follows:

* __AWS EC2__, per instructions at http://docs.docker.com/machine/drivers/aws/, the example given has 3 required parameters, 2 of which I'm not sure how to get:
 --amazonec2-access-key: access key id for the Amazon Web Services API
--amazonec2-secret-key: secret access key for the Amazon Web Services API.
The other required parameter is the VPC ID which is easy to get.

     I have a .pem file I generated for AWS, but this doesn't seem to fit with the example. There is a fingerprint for the key-pair, I'm wondering if this is one of the required keys? In general, the driver instructions, option descriptions, and how they map to the AWS EC2 Console are not intuitive.  I need to flesh out the content beyond these questions, but will need your help.

* __Microsoft Azure__, per instructions at http://docs.docker.com/machine/drivers/azure/ (and a few other sources) require that you upload a mycert.cer file (Settings > Management Certificates is one navigation path given). There is just no option I've found anywhere in their portal to upload a certificate file or any mention of certs at all. When you create the VM entirely through their portal, you have the option to provide your own password or SSH key.

Sorry for such detail in this comment, but it seemed like the best place to document the issues. As a prerequisite to re-writing these topics, I think it's imperative to clearly identify the install process Docker Machine on each platform. It's likely that part of the problem is aspects of 3rd party UI's changed somewhat since these topics were written. Let me know when you all or someone has some time to work past these problems with me.--Thanks!
